src/datanode/src/heartbeat/handler/flush_region.rs

```diff
@@ -22,102 +22,186 @@ use crate::error;
-                            row_group_size: None,
-                        });
-                        region_server_moved
-                            .handle_request(region_id, request)
-                            .await?;
-                        Ok(())
-                    }),
-                )
-                .await;
-            if register_result.is_busy() {
-                warn!("Another flush task is running for the region: {region_id}");
+            for region_id in region_ids.iter() {
+                match self.flush_single_region(*region_id).await {
+                    Some(InstructionReply::FlushRegion(SimpleReply {
+                        result: false,
+                        error: Some(err),
+                    })) => {
+                        errors.push(format!("Region {}: {}", region_id, err));
+                    }
+                    Some(_) => {
+                        // This case shouldn't happen as flush_single_region returns None on success
+                        errors.push(format!("Region {}: unexpected reply format", region_id));
+                    }
+                    None => {
+                        // Success case - flush_single_region returns None on success
+                        success_count += 1;
+                    }
+                }
             }
-            let mut watcher = register_result.into_watcher();
-            let result = self.flush_tasks.wait_until_finish(&mut watcher).await;
-            match result {
-                Ok(()) => Some(InstructionReply::FlushRegion(SimpleReply {
+
+            if errors.is_empty() {
+                // All regions flushed successfully
+                Some(InstructionReply::FlushRegion(SimpleReply {
                     result: true,
                     error: None,
-                })),
-                Err(err) => Some(InstructionReply::FlushRegion(SimpleReply {
-                    result: false,
-                    error: Some(format!("{err:?}")),
-                })),
+                }))
+            } else {
+                // Some regions failed, return aggregated error
+                let error_msg = if success_count > 0 {
+                    format!(
+                        "Partial success: {} regions flushed successfully, {} failed. Errors: [{}]",
+                        success_count,
+                        errors.len(),
+                        errors.join("; ")
+                    )
+                } else {
+                    format!("All regions failed. Errors: [{}]", errors.join("; "))
+                };
+                Some(InstructionReply::FlushRegion(SimpleReply {
+                    result: success_count > 0, // true if at least one region succeeded
+                    error: Some(error_msg),
+                }))
             }
-        })
+        }
+    }
+
+    async fn flush_single_region(&self, region_id: RegionId) -> Option<InstructionReply> {
```

Comment:
Can you use the `result = true/false` to indicate the flush result, instead of `Some/None`? I think it makes codes more understandable.

```suggestion
    async fn flush_single_region(&self, region_id: RegionId) -> InstructionReply {
```